src/components/Gallery/__tests__/Gallery.test.js

```diff
@@ -0,0 +1,70 @@
+/* eslint-disable sonarjs/no-duplicate-string */
+import React from 'react';
+import renderer from 'react-test-renderer';
+import { cleanup, render, fireEvent, waitFor } from '@testing-library/react';
+
+import '@testing-library/jest-dom';
+
+import Gallery from '../Gallery';
+
+const mockGalleryAssets = [
+  { src: 'https://placeimg.com/640/480/any' },
+  { src: 'https://placeimg.com/640/480/any' },
+  { src: 'https://placeimg.com/640/480/any' },
+  { src: 'https://placeimg.com/640/480/any' },
+  { src: 'https://placeimg.com/640/480/any' },
+  { src: 'https://placeimg.com/640/480/any' },
+];
+
+afterEach(cleanup); // eslint-disable-line
+
+describe('Gallery', () => {
+  it('should render component with default props', () => {
+    const tree = renderer
+      .create(<Gallery images={mockGalleryAssets.slice(0, 2)} />)
+      .toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it('should render component with 3 images', () => {
+    const tree = renderer
+      .create(<Gallery images={mockGalleryAssets.slice(0, 3)} />)
+      .toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it('should render component with 4 images', () => {
+    const tree = renderer
+      .create(<Gallery images={mockGalleryAssets.slice(0, 4)} />)
+      .toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it('should render component with 6 images', () => {
+    const tree = renderer
+      .create(<Gallery images={mockGalleryAssets} />)
+      .toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it('should open modal on image click', async () => {
```

Comment:
an extra possible test is having the modal open and close it by firing a click event outside of the doc, more of behavioral testing for the lib we using to be sure it's not broken on future versions

another one is snapshotting an open modal